inject.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlDecoder "k8s.io/apimachinery/pkg/util/yaml"
+
+	// https://github.com/kubernetes/client-go/blob/master/INSTALL.md
+	"k8s.io/client-go/pkg/api/v1"
+	batch "k8s.io/client-go/pkg/apis/batch/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+/*
+	Inject in init-container to set up transparent proxying
+	via iptables rules which send traffic to daemonset linkerd
+
+	This file based on:
+	https://github.com/istio/pilot/blob/master/platform/kube/inject/inject.go
+	and https://github.com/istio/pilot/tree/master/cmd/istioctl/inject.go
+
+  This code inserts the following into the pod specs:
+
+   spec:
+     initContainers:
+       - name: init
+         image: buoyantio/istio-init:v1
+         args:
+           - -p
+           - "4140"
+         imagePullPolicy: Always
+         securityContext:
+           capabilities:
+             add:
+             - NET_ADMIN
+*/
+
+const (
+	istioSidecarAnnotationSidecarKey   = "alpha.istio.io/linkerd-daemonset"
+	istioSidecarAnnotationSidecarValue = "injected"
+	initContainerName                  = "init-linkerd"
+	initImage                          = "buoyantio/istio-init:v1"
+)
+
+type Params struct {
+	LinkerdDaemonsetPort string
+}
+
+func dieIf(err error) {
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+}
+
+func injectIntoPodTemplateSpec(p *Params, t *v1.PodTemplateSpec) error {
+	// from https://github.com/istio/pilot/blob/master/platform/kube/inject/inject.go
+
+	if t.Annotations == nil {
+		t.Annotations = make(map[string]string)
+	} else if _, ok := t.Annotations[istioSidecarAnnotationSidecarKey]; ok {
+		// Return unmodified resource if init has already happened
+		return nil
+	}
+	t.Annotations[istioSidecarAnnotationSidecarKey] = istioSidecarAnnotationSidecarValue
+
+	// init-container
+	var annotations []interface{}
+	if initContainer, ok := t.Annotations["pod.beta.kubernetes.io/init-containers"]; ok {
+		if err := json.Unmarshal([]byte(initContainer), &annotations); err != nil {
+			return err
+		}
+	}
+	initArgs := []string{
+		"-p", p.LinkerdDaemonsetPort,
+	}
+	annotations = append(annotations, map[string]interface{}{
+		"name":            initContainerName,
+		"image":           initImage,
+		"args":            initArgs,
+		"imagePullPolicy": "Always",
+		"env": []map[string]interface{}{
+			map[string]interface{}{
+				"name": "NODE_NAME",
+				"valueFrom": map[string]interface{}{
+					"fieldRef": map[string]interface{}{
+						"apiVersion": "v1", // https://github.com/kubernetes/kubernetes/issues/39189
+						"fieldPath":  "spec.nodeName",
+					},
+				},
+			},
+		},
```

Comment:
Could you add this env section to the comment at the top of the file?

I'm a little concerned that this means this doesn't work in minikube? It's not a blocker for this initial PR, but I definitely am interested in finding a way to support minikube.